src/async.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use futures::Stream;
+use futures::stream::FusedStream;
+use futures::channel::mpsc::UnboundedReceiver;
+use futures::channel::mpsc::UnboundedSender;
+use futures::task::Poll;
+use ipc::IpcReceiver;
+use ipc::IpcReceiverSet;
+use ipc::IpcSelectionResult;
+use ipc::IpcSender;
+use ipc::OpaqueIpcMessage;
+use ipc::OpaqueIpcReceiver;
+use ipc;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::pin::Pin;
+use std::sync::Mutex;
+use std::task::Waker;
+use std::thread;
+
+/// A stream built from an IPC channel.
+pub struct IpcStream<T>(UnboundedReceiver<OpaqueIpcMessage>, PhantomData<T>);
+
+impl<T> Unpin for IpcStream<T> {}
+
+// A router which routes from an IPC channel to a stream.
+struct Router {
```

Comment:
`Router` and `ROUTER` are used in `router.rs`. Would it make sense to rename these to `AsyncRouter` and `ASYNC_ROUTER`?

Reply:
It's `AsyncRouter` vs `async::Router` which I don't have strong opinions about.